packages/driver/src/cy/multidomain/index.ts

```diff
@@ -105,8 +126,29 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
       communicator.on('command:enqueued', addCommand)
       communicator.on('command:update', updateCommand)
 
-      return new Bluebird((resolve) => {
-        communicator.once('ran:domain:fn', resolve)
+      return new Bluebird((resolve, reject) => {
+        communicator.once('run:domain:fn', (err) => err ? reject(err) : resolve())
+
+        communicator.once('unbind:done:called', () => {
+          // If `done()` is passed into the test, but is called in the primary domain and NOT the secondary domain
+          // Go ahead and make sure the listener is removed
+          communicator.off('done:called', invokeDone)
+
+          // If done was NOT called in the secondary domain, go ahead and make sure this promise is settled.
+          // if done was called in the secondary domain, this promise should already be settled and resolving it has no effect.
+          deferredDone.resolve()
+        })
+
+        // By the time the command queue is finished, this promise should be settled as
+        // as done will be invoked within the secondary domain already, if applicable
+        deferredDone.promise.then(({ crossDomainDoneCalled, error } = {
+          crossDomainDoneCalled: false,
+          error: undefined,
+        }) => {
+          if (crossDomainDoneCalled) {
+            done(error)
+          }
+        })
```

Comment:
What's the reason for using a deferred? Couldn't we just invoke `done` directly from the `done:called` handler and it would achieve the same thing?

Reply:
Fixed in b2def60cb0fe7cd087968d96c738676e70d3ef32